components/script/dom/bluetooth.rs

```diff
@@ -2,18 +2,34 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use core::clone::Clone;
 use dom::bindings::codegen::Bindings::BluetoothBinding;
-use dom::bindings::codegen::Bindings::BluetoothBinding::BluetoothMethods;
+use dom::bindings::codegen::Bindings::BluetoothBinding::RequestDeviceOptions;
+use dom::bindings::codegen::Bindings::BluetoothBinding::{BluetoothScanFilter, BluetoothMethods};
 use dom::bindings::codegen::Bindings::BluetoothDeviceBinding::VendorIDSource;
+use dom::bindings::codegen::UnionTypes::StringOrUnsignedLong;
+use dom::bindings::error::Error::Type;
+use dom::bindings::error::Fallible;
 use dom::bindings::global::GlobalRef;
 use dom::bindings::js::Root;
 use dom::bindings::reflector::{Reflectable, Reflector, reflect_dom_object};
 use dom::bluetoothadvertisingdata::BluetoothAdvertisingData;
 use dom::bluetoothdevice::BluetoothDevice;
+use dom::bluetoothuuid::BluetoothUUID;
 use ipc_channel::ipc::{self, IpcSender};
+use net_traits::bluetooth_scanfilter::BluetoothScanfilter;
+//use net_traits::bluetooth_scanfilter::{BluetoothScanfilter, BluetoothScanfilterSequence};
+use net_traits::bluetooth_scanfilter::{RequestDeviceoptions, ServiceUUIDSequence};
 use net_traits::bluetooth_thread::{BluetoothMethodMsg, BluetoothObjectMsg};
 use util::str::DOMString;
 
+const MAX_DEVICE_NAME_LENGTH: usize = 248;
+const MAX_FILTER_NAME_LENGTH: usize = 29;
+const NAME_TOO_LONG_EXCEPTION: &'static str = "A device name can't be longer than 248 bytes.";
+const FILTER_NAME_TOO_LONG_EXCEPTION: &'static str =
+    "A 'name' or 'namePrefix' longer than 29 bytes results in no devices being found, because
+ a device can't advertise a name longer than 29 bytes.";
```

Comment:
In the spec there is no restriction gor this message, I only used the solution I found in the chromium.